### PR TITLE
Guardian/Parasite Fixes

### LINKED ISSUE #12341: Guardian Battlecry can only accept 5 characters
Despite saying it takes 6

### LINKED ISSUE #12342: Guardian chat logs as "G"
No matter what message you put

Holoparasite Neon: G

Etc

### LINKED ISSUE #12343: Possible to kill a Guardian without killing the user
If gib() is called on the guardian (such as running into an emagged gibber), or they get hit by a deathbolt or whatever, it's possible to kill them without damage being passed to the user

### LINKED ISSUE #12351: Holoparasites take explosion damage while INSIDE the host
Happened yesterday: We were just about the end of the radius of damage of a sci bomb, so the damage dealt to us was very little: almost negligible. What wasn't negligible, however, was the red text:

"Blood sprays from [my host] as [I] take damage."

A borg saw it, called security, and that was that.

Evidently, if a holoparasite is inside his host, the holoparasite himself should not be taking damage and displaying "my-host-is-a-traitor" messages around.

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Fixes Guardian logging.

Fixes Guardians gibbing without killing their owner

Fixes Guardians taking damage while recalled

Fixes Guardian input length being wrong

Fixes #12342
Fixes #12341
Fixes #12343
Fixes #12351

### CODE REVIEW

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 1]
```diff
@@ -130,7 +135,7 @@
 		if(M == src.summoner || (M in dead_mob_list))
 			M << "<span class='boldannounce'><i>[src]:</i> [input]</span>"
 	src << "<span class='boldannounce'><i>[src]:</i> [input]</span>"
-	log_say("[src.real_name]/[src.key] : [text]")
+	log_say("[src.real_name]/[src.key] : [input]")
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: Ha, this is brill
```text``` is the byond var that defaults to the first letter of the atom's name, and it's used for the ascii/text view mode.

rofl

Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: no bully

Comment (Dorsisdwarf) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: Someone should buff ascii mode

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 2]
```diff
@@ -91,6 +90,12 @@
 		if(3)
 			adjustBruteLoss(30)
 
+/mob/living/simple_animal/hostile/guardian/gib()
+	if(summoner)
+		src.summoner << "<span class='danger'><B>Your [src.name] was blown up!</span></B>"
```
Comment (duncathan) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: >`[src.name]`

The first thing wrong with this is that prefixing local variables with src is dumb if there isn't potential scope confusion
The second thing wrong is the fact that you're embedding the name at all; you just need to embed src and it does that for you
Make it `Your [src]`

Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: I mostly do it for myself, I just prefer to have that on there for whatever reason. I started removing them but realized how many there were and how late it was.

**Pull request merged**